internal/core/component_factory.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+#![warn(missing_docs)]
+
+//! This module contains the `ComponentFactory` and a `Component`. Both are
+//! exposed to the C API.
+
+use core::ffi::c_void;
+use core::fmt::Debug;
+
+use alloc::boxed::Box;
+use alloc::rc::Rc;
+
+use crate::{
+    api::ComponentHandle,
+    component::{ComponentRc, ComponentVTable},
+};
+
+/// A `Component` built by a `ComponentFactory`
+#[repr(C)]
+pub struct Component(ComponentRc);
```

Comment:
Naming wise this could be called `Product` instead of yet another `Component`. Factories build products.

Ideally, I think this type would not be public but merely pub(crate), or is there going to be a factory consumer outside of core?

Reply:
I think `Component` is right. Because that's how we call it in your slint file.

Reply:
`component` shows up in slint when you do

```
component Foo {
   // ...
} 
```

Reply:
but the `build` function doesn't need to be public